File: pset1/cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int getCoins(int);

int main(void){
    float change;
    int cents;
    do{
       change = get_float("Change owed: ");
    }while (change <= 0);

    cents = round(change * 100);

    printf("%i\n", getCoins(cents));
}

int getCoins(int cents)
{
    int coins = 0;  
    while (cents != 0)
    {
        if (cents >= 25)
        {
            cents -= 25;
            coins++;
        }
        else if (cents >= 10)
        {
            cents -= 10;
            coins++;
        }
        else if (cents >= 5)
        {
            cents -= 5;
            coins++;
        }
        else
        {
            cents--;
            coins++;
        }
    }

    return coins;
}

File: pset6/cash/cash.py
from cs50 import get_float


def main():
    n = 0
    while True:
        n = get_float("Change owed: ")
        if not n < 0:
            break
    n = int(round(n * 100))  # dollar to cent
    print(number_of_coins(n))


def number_of_coins(n):
    coins = 0
    while not n == 0:
        if n >= 25:
            n -= 25
            coins += 1
        elif n >= 10:
            n -= 10
            coins += 1
        elif n >= 5:
            n -= 5
            coins += 1
        elif n >= 1:
            n -= 1
            coins += 1

    return coins


if __name__ == "__main__":
    main()